### ISSUE TITLE
Multilanguage support

### ISSUE THREAD

**Issue by xen2** [dotnet/docfx#803]
Is there any plan for multilanguage support?

That's probably something that could work per document, for example, MyPage.md and MyPage.jp.md
(It could use default version as a fallback if no translation present)

**Comment by xen2** [comment]
Note: if nothing is planned, we'll probably just build non-english doc by taking english one, overwrite the few files that are different and regenerate a full doc under a different URL (i.e. /jp instead of /en).
That way, it will keep english doc for everything not translated.

**Comment by FSharpCSharp** [comment]
In my opinion, the topic of "multilingualism" is essential. Is there any ideas in doxFX how to best represent this? We have a similar requirement here that we need to provide help in different languages. Or is there Microsoft "Best Practices", which should be considered?

**Comment by zwcloud** [comment]
@FSharpCSharp 
The Microsoft docs uses multiple docfx sites for multilingualism, such as [docs(en-us)](https://github.com/dotnet/docs), [docs.zh-cn](https://github.com/dotnet/docs.zh-cn) and so on.
https://github.com/dotnet?q=docs

**Comment by FSharpCSharp** [comment]
@zwcloud 
That's right! But apparently Microsoft also has a special docfx template to switch the languages easily. This is unfortunately not included in the standard template, which makes it very difficult to use from my point of view. I agree that it would of course be best to benefit from the templates of Microsoft docs.

**Comment by fleed** [comment]
@FSharpCSharp there are many advanced features from Microsoft docs that should be added to the standard template, or released as advanced template. Hopefully something will hear us..

**Comment by superyyrrzz** [comment]
@fleed docs.msft have some extra feature on hosting layer to make multi-lang happen. However, it might not be the same implementation for a static site generator like DocFX. It still needs some design work.

**Comment by fleed** [comment]
@superyyrrzz I'm also implementing a .NET core web app with a few middlewares to wrap the statically generated website.. but I feel like "reinventing the wheel"

**Comment by superyyrrzz** [comment]
@FSharpCSharp @fleed We haven't plan to support it in v2. I'm afraid it will not come soon. 

Now we are designing v3. I removed it from board as it is not the top priority for now, but I have added it to the [commnity asks](https://github.com/dotnet/docfx/projects/1#card-8543119). This is a valid user scenario we aim to support.

It is also welcome to share your expected input/output structure to help us better understand the request. Our best practice for docs.ms.com may not apply for other open source repos, as it heavily relies on the system behind DocFX, which is hard to open source.

**Comment by FSharpCSharp** [comment]
@superyyrrzz 
Thank you very much for the detailed answer. I find it reassuring to hear that the issue has not disappeared, but is planned for the long-term future. In my opinion, this is also essential because more and more Microsoft products are using docfx, and users are virtually forced to provide documentation or user support in similar ways.

Thank you very much for your commitment, and so on!

**Comment by OsmondJiang** [comment]
#3343  and #3341

**Comment by FSharpCSharp** [comment]
@OsmondJiang 
It's really great that things are moving forward in this area, and a design document is already available from Microsoft! Thank you for accepting this topic.

**Comment by sindharta** [comment]
I saw that localization was supposed to be added in Q1 2019 in this document: 
https://github.com/dotnet/docfx/projects/1,
but the card is still in TODO state, and the project itself seems to be "closed".

Any idea when this feature (and v3 in general) is ready for release ?

**Comment by yufeih** [comment]
@sindharta , localization feature is implemented in v3. But v3 currently isn't generally available for public release due to other reasons, 

The project on GitHub is __closed__ because we've switched to azure devops for workitem tracking.

**Comment by sindharta** [comment]
@yufeih I see. Is there any ETA when we can expect v3 to be released ?

**Comment by zwcloud** [comment]
> we've switched to azure devops for workitem tracking.
@yufeih 
Where is it? I cannot see that.
https://docfx.visualstudio.com/docfx/_workitems/recentlyupdated

![image](https://user-images.githubusercontent.com/6965758/64503281-d940b980-d2fc-11e9-9e22-cc63773ec8c4.png)

**Comment by yufeih** [comment]
@zwcloud, it is a private azure devops project to correlate with other workitems for https://docs.microsoft.com that are close sourced.

**Comment by fleed** [comment]
@yufeih is [https://github.com/dotnet/docfx/blob/v3/docs/roadmap.md](https://github.com/dotnet/docfx/blob/v3/docs/roadmap.md) the official place to track the progress?

**Comment by yufeih** [comment]
@sindharta @fleed , yes, https://github.com/dotnet/docfx/blob/v3/docs/roadmap.md is currently the official place to track progress.

**Comment by chri5topherus** [comment]
Are there any updates regarding the multilanguage support? Haven't not found any documentation.
